### PR TITLE
[netcore] Onboard Arcade

### PR DESCRIPTION
**Pull request by directhex**
Work in progress for integrating into dnceng practices.

The pile of stuff in eng/common is verbatim from https://github.com/dotnet/arcade-minimalci-sample and at some point will get auto-PR'd with changes.

empty.proj and publishwitharcade.proj are hacks from coreclr

I know adding things to top-level sucks. It's unavoidable with some of the tooling, which requires it.

The SHOUTY VARIABLES in publishwitharcade.proj are VSTS defaults, hopefully they're populated from the environment.

Starting point for documentation is https://github.com/dotnet/arcade/blob/master/Documentation/Onboarding.md

I don't think I have the BAR stuff right yet.

### CODE REVIEW

File: scripts/ci/pipeline-netcore-runtime.yml [hunk 1]
```diff
@@ -4,7 +4,15 @@ pool:
 variables:
 - ${{ if ne(variables['System.TeamProject'], 'public') }}:
   - group: DotNet-VSTS-Bot
-
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), ne(variables['Build.Reason'], 'PullRequest')) }}:
+    # Variables used to publish packages to blob feed
+    - name: dotnetfeedUrl
+      value: https://dotnetfeed.blob.core.windows.net/dotnet-coreclr/index.json
+    - name: dotnetfeedPAT
```
Comment (chcosta) [inline scripts/ci/pipeline-netcore-runtime.yml]: nit: I understand casing in most of the yml file examples isn't consistent, but it'd be nice to move towards some consistency here.  ie, I see things named with all kinds of casing / naming patterns...

- dotnetfeedUrl (yaml specific variable)
- dotnetfeedPAT (yaml specific variable but with different casing)
- dotnetfeed-storage-access-key-1 (different casing again - a victim of the variable name in key vault)
- _DotNetPublishToBlobFeed (different casing again - a victim of what the Arcade Sdk expects)

I guess my nit is really, can we change this to `dotnetFeedPat` and `dotnetFeedUrl`?

File: eng/publishwitharcade.proj [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  <PropertyGroup>
+    <!-- Set the TargetFramework just to make the SDK happy -->
+    <TargetFramework>net462</TargetFramework>
+  </PropertyGroup>
+
+  <Target Name="PublishPackages">
+
+    <ItemGroup>
+      <PackagesToPublish Include="artifacts\*.nupkg" />
```
[the remark this answers is not in the document]
Comment (directhex) [reply eng/publishwitharcade.proj]: At this point, yes, it's intentional. The underlying Automake project doesn't really distinguish between any of this stuff - we either produce a nupkg to use, or we don't. Maybe down the line we'll care more.

**Comment by directhex** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=175389

I mean, it broke, but that looks pretty close to functional?

**Comment by directhex** [comment]
I think the break was from a previous build - since I keep force-pushing a single commit, the build number (therefore package version) never changes, so a previous failure might be the cause of the "already exists" error

**Comment by directhex** [comment]
@monojenkins build failed

**Comment by directhex** [comment]
(right now, I'm focused on Linux x64 because that's my wheelhouse - we can worry about a bigger matrix later, once we're confident we can produce artifacts and consume them in core-setup)

**Comment by directhex** [comment]
@monojenkins build failed

**Pull request merged**